rules/doctrine-code-quality/src/Rector/Property/RemoveRedundantDefaultAnnotationValuesRector.php

```diff
@@ -0,0 +1,250 @@
+ */
+final class RemoveRedundantDefaultAnnotationValuesRector extends AbstractRector
+{
+    /**
+     * @var DoctrinePropertyAnalyzer
+     */
+    private $doctrinePropertyAnalyzer;
+
+    /**
+     * @var DoctrineClassAnalyzer
+     */
+    private $doctrineClassAnalyzer;
+
+    /**
+     * @var bool
+     */
+    private $hasModifiedAnnotation = false;
+
+    public function __construct(
+        DoctrinePropertyAnalyzer $doctrinePropertyAnalyzer,
+        DoctrineClassAnalyzer $doctrineClassAnalyzer
+    ) {
+        $this->doctrinePropertyAnalyzer = $doctrinePropertyAnalyzer;
+        $this->doctrineClassAnalyzer = $doctrineClassAnalyzer;
+
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Removes redundant default values from Doctrine ORM annotations.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\ManyToOne(targetEntity=Training::class)
+     * @ORM\JoinColumn(name="training", unique=false)
+     */
+    private $training;
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity
+ */
+class SomeClass
+{
+    /**
+     * @ORM\ManyToOne(targetEntity=Training::class)
+     * @ORM\JoinColumn(name="training")
+     */
+    private $training;
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class, Property::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        $this->hasModifiedAnnotation = false;
+        if ($node instanceof Property) {
+            $this->refactorPropertyAnnotations($node);
+        }
+
+        if ($node instanceof Class_) {
+            $this->refactorClassAnnotations($node);
+        }
+
+        if (! $this->hasModifiedAnnotation) {
+            return null;
+        }
+
+        return $node;
+    }
+
+    private function refactorPropertyAnnotations(Property $node): void
+    {
+        $this->refactorColumnAnnotation($node);
+        $this->refactorGeneratedValueAnnotation($node);
+        $this->refactorJoinColumnAnnotation($node);
+        $this->refactorManyToManyAnnotation($node);
+        $this->refactorManyToOneAnnotation($node);
+        $this->refactorOneToManyAnnotation($node);
+        $this->refactorOneToOneAnnotation($node);
+    }
+
+    private function refactorColumnAnnotation(Property $node): void
+    {
+        $columnTagValueNode = $this->doctrinePropertyAnalyzer->matchDoctrineColumnTagValueNode($node);
+        if ($columnTagValueNode === null) {
+            return;
+        }
+
+        $this->removeItemWithDefaultValue($columnTagValueNode, 'nullable', false);
```

Comment:
This logic can be decouple to own services, that would be shared by 2 rules

Reply:
Done 👍